src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -61,53 +64,58 @@ public int compare(Person o1, Person o2) {
 		});
 		// TODO использовать Arrays.sort
 		
-		assertArrayEquals(
-				new Person[]{new Person("Алексей", "Доренко", 40), new Person("Артем", "Зимов", 45),
-							 new Person("Николай", "Зимов", 30),
-							 new Person("Иван", "Мельников", 20)}, persons);
+		assertArrayEquals(new Person[]{
+				new Person("Алексей", "Доренко", 40),
+				new Person("Артем", "Зимов", 45),
+				new Person("Иван", "Мельников", 20),
+				new Person("Николай", "Зимов", 30)}, persons);
 	}
 	
 	@Test
 	public void findFirstWithAge30UsingGuavaPredicate() {
 		List<Person> persons = Arrays.asList(getPersons());
-		
-		Optional<Person> personOptional;
-		
-		if ((personOptional =
-					 FluentIterable.from(persons).firstMatch(input -> (input.getAge() == 30)))
-				.isPresent()) {
-			System.out.println(personOptional.get());
-			assertEquals(new Person("Алексей", "Доренко", 40), personOptional.get());
-		}
+		Predicate personPredicate = new Predicate<Person>() {
+			@Override
+			public boolean apply(Person inputPerson) {
+				return (inputPerson.getAge() == 30);
+			}
+		};
+		Optional<Person> personOptional = FluentIterable.from(persons).firstMatch(personPredicate);
+		Person person = personOptional.get();
+		System.out.println(personOptional.get());
+		assertEquals(new Person("Николай", "Зимов", 30), person);
 		// TODO использовать FluentIterable
-		Person person = null;
 		
-		assertEquals(new Person("Николай", "Зимов", 30), person);
 	}
 	
 	@Test
 	public void findFirstWithAge30UsingGuavaAnonymousPredicate() {
 		List<Person> persons = Arrays.asList(getPersons());
 		
-		
-		Optional<Person> personOptional = FluentIterable.from(persons).firstMatch(
-				person -> person.equals(person.getFirstName()));
-		
+		Optional<Person> personOptional =
+				FluentIterable.from(persons).firstMatch((new Predicate<Person>() {
+					@Override
+					public boolean apply(Person inputPerson) {
+						return (inputPerson.getAge() == 30);
+					}
+				}));
 		if (personOptional.isPresent()) {
 			System.out.println(personOptional.get());
-			assertEquals(new Person("Алексей", "Доренко", 40), personOptional.get());
+			assertEquals(new Person("Николай", "Зимов", 30), personOptional.get());
 		}
 		// TODO использовать FluentIterable
-		Person person = null;
+		Person person = personOptional.get();
 		
 		assertEquals(new Person("Николай", "Зимов", 30), person);
 	}
 	
 	private Person[] getPersons() {
 		
 		
-		return new Person[]{new Person("Иван", "Мельников", 20),
-							new Person("Алексей", "Доренко", 40),
-							new Person("Николай", "Зимов", 30), new Person("Артем", "Зимов", 45)};
+		return new Person[]{
+				new Person("Иван", "Мельников", 20),
+				new Person("Николай", "Зимов", 30),
+				new Person("Алексей", "Доренко", 40),
+				new Person("Артем", "Зимов", 45)};
```

Comment:
in my point of view you should sort array of persons in the methods, not change the task

Reply:
+